### PR TITLE
upgrade test: retry truncate and ignore the truncate timeout error

### PR DESCRIPTION
**Pull request by amoskong**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by amoskong** [comment]
Need backport this PR to upgrade-3.1

File: sdcm/fill_db_data.py [hunk 1]
```diff
@@ -2891,12 +2893,17 @@ def cql_create_tables(self, session):
         for truncate in truncates:
             session.execute(truncate)
 
+    @retrying(n=3, sleep_time=20, allowed_exceptions=ProtocolException)
+    def truncate_table(self, session, truncate):  # pylint: disable=no-self-use
+        LOGGER.debug(truncate)
+        session.execute(truncate)
+
     def truncate_tables(self, session):
         # Run through the list of items and create all tables
```
Comment (yarongilor) [inline sdcm/fill_db_data.py]: why it is written "create all tables" ? isn't it only truncates tables?

File: sdcm/fill_db_data.py [hunk 2]
```diff
@@ -2891,12 +2893,17 @@ def cql_create_tables(self, session):
         for truncate in truncates:
             session.execute(truncate)
 
+    @retrying(n=3, sleep_time=20, allowed_exceptions=ProtocolException)
+    def truncate_table(self, session, truncate):  # pylint: disable=no-self-use
```
Comment (yarongilor) [inline sdcm/fill_db_data.py]: why is 'no-self-use' needed here? why not making it static without 'self'?

File: sdcm/fill_db_data.py [hunk 3]
```diff
@@ -2972,7 +2979,10 @@ def fill_db_data(self):
             session.default_consistency_level = ConsistencyLevel.QUORUM
             session.set_keyspace("keyspace_fill_db_data")
             # clean original test data by truncate
-            self.truncate_tables(session)
+            try:
```
Comment (yarongilor) [inline sdcm/fill_db_data.py]: not sure this general try-except is needed. perhaps it's good enough without it.

**Comment by roydahan** [comment]
Need to backport to 3.3

File: sdcm/fill_db_data.py (hunk 1 above)
Comment (amoskong) [reply sdcm/fill_db_data.py]: You are right, I will fix the comment later

File: sdcm/fill_db_data.py (hunk 3 above)
Comment (amoskong) [reply sdcm/fill_db_data.py]: If the last retry fails, exception will raise and break the subtest, that's not expected.

**Pull request merged**